worlds/alttp/Options.py

```diff
@@ -122,22 +126,151 @@ class TriforcePieces(Range):
     range_end = 90
 
 
+class TriforceMode(Choice):
+    """Determines how extra available triforce pieces are calculated. Up to 90 can be available.
+    Extra makes available pieces = extra + required.
+    Percentage makes available pieces = percentage * required.
+    Available makes available pieces = available value."""
+    display_name = "Triforce Pieces Mode"
+    option_extra = 0
+    option_percentage = 1
+    option_available = 2
+    default = option_available
+
+
+class TriforceExtra(TriforcePieces):
+    """How many extra Triforce pieces are available."""
+    display_name = "Extra Triforce Pieces"
+    default = 10
+
+
+class TriforcePercentage(Range):
+    """Percentage of required Triforce pieces that will be available."""
```

Comment:
probably for a separate PR, but at some point we should have a `display_if = {"goal": ("trifoce hunt",)}` or something that hides the option in the UI if the requirement is not statisfied.

Reply:
i think the best way would be having some sort of `option_requires` where you can define an option and a specified value for said option. This would allow for webhost options to dynamically gray themselves out etc too.